src/components/token/Tokens.js

```diff
@@ -24,18 +27,101 @@ class Tokens extends React.Component {
             tokens: [],
             hasAfter: false,
             hasBefore: false,
-            searchText: ""
+            searchText: "",
+            sortBy: "uid",
+            order: "asc",
+            page: 1,
+            //This variable stores the pages and its searchAfter params that is passed to explorer-service
+            pageSearchAfter: [],
+            loading: true
         }
     }
 
-    onSearchButtonClicked = () => {
-        // TODO: Implement behavior when search button is clicked
+    componentDidMount = async() => {
+        //"Click" on search to make the first query
+        await this.onSearchButtonClicked();
+        this.setState({loading: false})
+    }
+
+    resetPageSearchAfter = () => {
+        this.setState({pageSearchAfter: [
+            {
+                page: 1,
+                searchAfter: []
+            }
+        ]})
+    }
+
+    getTokens = async (searchAfter) => {
+        const tokensRequest = await tokensApi.getList(this.state.searchText, this.state.sortBy, this.state.order, searchAfter)
+        let tokens = get(tokensRequest, 'data', {hits:[], 'has_next': false})
+        forEach(tokens.hits, (token) => {
+            if(isNil(token.uid)) {
```

Comment:
The backend is sending sometimes `uid` and sometimes `id`?

Reply:
Backend is sending as `id`. And as we are using `id` and `uid` as interchangeable terms in this context, I added this verification to make sure we get the value independently of what we receive from backend.